CONTRIBUTING.md

```diff
@@ -1,19 +1,155 @@
+
+```sh
+yarn build:cli
+```
+
+To test the local version of the CLI, run `npx field-plugin <command>`. It is recommended to test the CLI outside of the Field Plugin SDK repository. To do this, run the following:
+
+```sh
+npx field-plugin create --dir ../<SOME-TEST-DIRECTORY>
+```
+
+A plugin will be created under the test directory.
+
+The CLI package currently has few unit tests, but you can execute them as follows:
+
+```sh
+yarn test:cli
+```
 
 ### Templates
 
-These are things to consider when creating a new template
+#### Testing existing templates
+
+If you want to try the React template, run this command:
+
+```sh
+yarn dev:react
+```
+
+You can also use the commands for other templates.
+
+- dev:react
+- dev:js
+- dev:vue2
+- dev:vue3
+
+#### Adding a new template
+
+We want to create a script to add a new template, but it's not available yet. So the process involves duplicating an existing template and modifying the code, particularly `vite.config.ts` for the build process.
+
+Things to keep in mind when creating a new template include:
+
+- Naming `gitignore` files without the dot (`gitignore`) as otherwise the file will be ignored by `npm publish`.
+- Omitting `.env.local.example` as it will be copied from the `monorepo` template.
+
+### Helpers
+
+This section will be filled once the helpers are shipped. Work is in progress.
+
+## Release
+
+You can release either `@storyblok/field-plugin` or `@storyblok/field-plugin-cli`. To begin the release process, run the following command on the `main` branch:
+
+```sh
+yarn bump-version
+```
+
+This will prompt you to select which package (`@storyblok/field-plugin` or `@storyblok/field-plugin-cli`) to release and the version number. After entering the required information, a pull request will be created automatically. This pull request will include changes in the `package.json` and possibly the `yarn.lock`.
+
+Once this pull request is reviewed and merged, you'll get a commit like [this](https://github.com/storyblok/field-plugin/commit/b4bd948ce3d26f0905352ddbe474ebc9e2f89159).
+
+Then, go to [Releases](https://github.com/storyblok/field-plugin/releases) and create a new release. The title should follow the format `<PACKAGE-NAME>@<VERSION>`. Examples:
+
+- `@storyblok/field-plugin@0.0.1`
+- `@storyblok/field-plugin-cli@1.0.0-beta.2`
+
+You can find a sample release [here](https://github.com/storyblok/field-plugin/releases/tag/%40storyblok%2Ffield-plugin-cli%400.0.1-beta.2). Make sure you write accurate changelogs, ensuring that library changes are not included in a CLI release and vice versa.
+
+Once a release is created, one of the two GitHub Actions - [.github/workflows/npm-publish-library.yml](https://github.com/storyblok/field-plugin/blob/main/.github/workflows/npm-publish-library.yml) or [.github/workflows/npm-publish-cli.yml](https://github.com/storyblok/field-plugin/blob/main/.github/workflows/npm-publish-cli.yml) - will run and deploy it to NPM.
+
+### Order of releases
```

Comment:
Should we also add a section explaining what is important for the PR. Like explaining what has been changes and some guidelines to not submit a huge PR or something similar